backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs

```diff
@@ -0,0 +1,46 @@
+﻿// ==========================================================================
+//  Squidex Headless CMS
+// ==========================================================================
+//  Copyright (c) Squidex UG (haftungsbeschränkt)
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System.Net;
+using System.Net.Http;
+using IdentityModel.Client;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Squidex.Config.Authentication
+{
+    public static class OAuth2AuthenticationServices
+    {
+        public static AuthenticationBuilder AddSquidexExternalOAuth2Authentication(this AuthenticationBuilder authBuilder, MyIdentityOptions identityOptions)
+        {
+            if (identityOptions.IsOAuthConfigured())
+            {
+                HttpClientHandler hch = new HttpClientHandler();
+                hch.ServerCertificateCustomValidationCallback = delegate { return true; };
+
+                HttpClient httpClient = new HttpClient(new LoggingHandler(hch));
+                httpClient.SetBasicAuthenticationOAuth(identityOptions.OAuthClientId, identityOptions.OAuthSecret);
```

Comment:
What is this? Is this part of the oauth specification? Why do you have to define it?

Reply:
The basic authentication is a part of the oauth specification. It's optional, I'll parametrize it.